pkg/crc/api/helpers.go

```diff
@@ -0,0 +1,113 @@
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+)
+
+type context struct {
+	method      string
+	requestBody []byte
+
+	code         int
+	headers      map[string]string
+	responseBody []byte
+}
+
+func (c *context) Bind(r interface{}) error {
+	return json.Unmarshal(c.requestBody, r)
+}
+
+func (c *context) JSON(code int, r interface{}) error {
+	c.code = code
+	var err error
+	c.responseBody, err = json.Marshal(r)
+	c.headers["Content-Type"] = "application/json"
+	return err
+}
+
+func (c *context) String(code int, r string) error {
+	c.code = code
+	var err error
+	c.responseBody = []byte(r)
+	c.headers["Content-Type"] = "text/plain; charset=UTF-8"
+	return err
+}
+
+type server struct {
+	routes     map[string]map[string]func(*context) error
+	routesLock sync.RWMutex
+}
+
+func newServer() *server {
+	return &server{
+		routes: make(map[string]map[string]func(*context) error),
+	}
+}
+
+func (s *server) GET(pattern string, handler func(c *context) error) {
+	if _, ok := s.routes[pattern]; !ok {
+		s.routes[pattern] = make(map[string]func(*context) error)
+	}
+	s.routes[pattern][http.MethodGet] = handler
+}
+
+func (s *server) POST(pattern string, handler func(c *context) error) {
+	if _, ok := s.routes[pattern]; !ok {
+		s.routes[pattern] = make(map[string]func(*context) error)
+	}
+	s.routes[pattern][http.MethodPost] = handler
+}
+
+func (s *server) DELETE(pattern string, handler func(c *context) error) {
+	if _, ok := s.routes[pattern]; !ok {
+		s.routes[pattern] = make(map[string]func(*context) error)
+	}
+	s.routes[pattern][http.MethodDelete] = handler
+}
+
+func (s *server) Handler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		s.routesLock.RLock()
```

Comment:
`GET`/`POST`/... should `WLock()` it, or you could remove `routesLock` altogether?